src/main/java/homework4/Bank.java

```diff
@@ -0,0 +1,41 @@
+package homework4;
+
+import java.util.HashMap;
+
+public class Bank<Integer, V extends Asset> {
```

Comment:
What reason of generic Integer there?

Reply:
Logic bag security. For example, if one programmer-user will put an Integer 123, and another will put an Long 123, there will be two records for the same account.

Reply:
I wrote in this case you don't need generic. You always have Integer. It is unlogic construction

Reply:
There is a use for Value, but it also can be replaced. The only true rison for generic was extention of Map

Reply:
No any relations between Bank\<Integer\,T> and internal Map\<Integer, T\>

"Don't use any collections in your homework. Look at HW other students. **I would like how you can work with generics.** You use there Map<k,v> logic" and I cannot undertand your knowledge about it